pkg/parse_hal/parse_hal_test.go

```diff
@@ -0,0 +1,279 @@
+/*
+ * Copyright The Spinnaker Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package parse_hal
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/spinnaker/kleat/api/client"
+)
+
+func TestEmptyHalConfigToClouddriver(t *testing.T) {
```

Comment:
I wonder if we'd ever want to refactor these as [table driven tests](https://github.com/golang/go/wiki/TableDrivenTests) just to emphasize what's actually different about each one and make it clear which cases we'd expect to be added for each new service. On the other hand we'd lose your very nice description of each case. 🤷

Reply:
That's a good thought!  Let me read about table driven tests!  These tests were mostly written by seeing what you did for the end-to-end tests in this repo then working from there, so I have a lot to learn about testing in go.